tests/allotrope/schema_parser/schema_cleaner_test.py

```diff
@@ -0,0 +1,1392 @@
+        ]
+    }
+    validate_cleaned_schema(
+        schema,
+        {
+            "anyOf": [
+                {
+                    "properties": {
+                        "key1": "otherValue",
+                        "key3": "value",
+                        "key4": "value",
+                    }
+                },
+                {
+                    "properties": {
+                        "key1": "value",
+                        "key2": "value",
+                        "key4": "value",
+                    }
+                },
+            ]
+        },
+    )
+
+
+def test_combine_anyof_with_multiple_conflicting_keys() -> None:
+    schema = {
+        "anyOf": [
+            {"properties": {"key1": "value", "key2": "value"}},
+            {"properties": {"key1": "otherValue", "key2": "otherValue"}},
+        ]
+    }
+    validate_cleaned_schema(
+        schema,
+        {
+            "anyOf": [
+                {"properties": {"key1": "otherValue", "key2": "otherValue"}},
+                {"properties": {"key1": "value", "key2": "value"}},
+            ]
+        },
+    )
+
+
+def test_combine_anyof_with_nested_conflicting_keys() -> None:
+    schema = {
+        "anyOf": [
+            {
+                "properties": {
+                    "obj1": {"properties": {"key1": "string"}},
+                    "field1": "value",
+                }
+            },
+            {
+                "properties": {
+                    "obj1": {"properties": {"key1": "boolean"}},
+                }
+            },
+        ]
+    }
+    validate_cleaned_schema(
+        schema,
+        {
+            "anyOf": [
+                {
+                    "properties": {
+                        "obj1": {"properties": {"key1": "boolean"}},
+                    }
+                },
+                {
+                    "properties": {
+                        "obj1": {"properties": {"key1": "string"}},
+                        "field1": "value",
+                    }
+                },
+            ]
+        },
+    )
+
+
+def test_combine_anyof_with_nested_anyof() -> None:
+    schema = {
+        "anyOf": [
+            {
+                "properties": {
+                    "obj1": {
+                        "anyOf": [
+                            {"properties": {"key1": "value"}},
+                            {"properties": {"key2": "value"}},
+                        ]
+                    }
+                }
+            },
+            {
+                "properties": {
+                    "obj1": {
+                        "anyOf": [
+                            {"properties": {"key3": "value"}},
+                            {"properties": {"key4": "value"}},
+                        ]
+                    }
+                }
+            },
+        ]
+    }
+    validate_cleaned_schema(
+        schema,
+        {
+            "properties": {
+                "obj1": {
+                    "allOf": [
```

Comment:
might be misunderstanding but why does this become an `allOf` here? (instead of staying an `anyOf`)